pkg/container/container.go

```diff
@@ -72,13 +96,121 @@ type container struct {
 	id         string
 	name       string
 	privileged bool
+	spec       generate.Generator
 }
 
 // New creates a new, empty Sandbox instance
-func New(ctx context.Context) Container {
+func New(ctx context.Context) (Container, error) {
+	spec, err := generate.New("linux")
+	if err != nil {
+		return nil, err
+	}
 	return &container{
-		ctx: ctx,
+		ctx:  ctx,
+		spec: spec,
+	}, nil
+}
+
+// SpecAddMount adds a specified mount to the spec
+func (c *container) SpecAddMount(destination, mntType, sourceType, mountToRemove string, options []string) {
+	c.spec.RemoveMount(mountToRemove)
```

Comment:
as pointed out above, I find the "mountToRemove" thing questionable.

Reply:
fixed!